app/controllers/users_controller.rb

```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  before_action :set_users, only: [:index]
+  before_action :set_user, only: [:show]
+
+  def index
+    # @users is already set by the `set_users` method
+  end
+
+  def show
+    # @user is already set by the `set_user` method
+  end
+
+  private
+
+  def set_users
+    @set_users ||= User.includes(:posts).all
+  end
+
+  def set_user
+    @set_user ||= User.includes(:posts).find(params[:id])
+    @recent_posts = @user.recent_posts
```

Comment:
- [ ] Please use correct method name `recent_posts` is undefined method so kindly fix this error as well :slightly_smiling_face: 

![Screenshot from 2023-11-26 02-41-56](https://github.com/BilalLiaquat7/blog_app/assets/45226848/ad6084f0-cc35-45e7-85da-c632ccfe1143)